wgpu-hal/src/gles/mod.rs

```diff
@@ -201,8 +213,16 @@ pub struct Buffer {
     target: BindTarget,
     size: wgt::BufferAddress,
     map_flags: u32,
+    #[cfg(target_arch = "wasm32")]
+    emulate_map_allocation: std::cell::Cell<Option<*mut u8>>,
```

Comment:
if the buffer is mappable, why not just create this as `Vec<>` at start and keep around?